dipy/segment/tissue.py

```diff
@@ -125,3 +125,98 @@ def classify(self, image, nclasses, beta, *, tolerance=1e-05, max_iter=100):
         PVE = PVE[..., 1:]
 
         return initial_segmentation, final_segmentation, PVE
+
+
+def compute_directional_average(data, bvals, b0_thresh=10):
+    """
+    Compute the mean signal for each unique b-value shell and fit a linear model.
+
+    Parameters
+    ----------
+        data : ndarray
+            The diffusion MRI data.
+        bvals : ndarray
+            The b-values corresponding to the diffusion data.
+
+    Returns
+    -------
+        P : float
+            The slope of the linear model.
+        V : float
+            The intercept of the linear model.
+    """
+    unique_bvals = np.unique(bvals)
+    # Mean signal for b=0 (non-diffusion weighted)
+    s0 = data[bvals < b0_thresh].mean()
+
+    if len(unique_bvals) <= 2:
+        raise ValueError("Insufficient unique b-values for fitting.")
+
+    # If the mean signal for b=0 is too low, return 0, 0
+    if data[bvals < b0_thresh].mean() < 50:
+        return 0, 0
+
+    S_bvals = [data[bvals == bval].mean() / (s0 + 0.01)
+               for bval in unique_bvals[1:]]
+    S_bvals = np.array(S_bvals)
+
+    # Avoid log(0)
+    if 0 in S_bvals:
+        S_bvals = S_bvals + 0.001
+
+    S_log = np.log(S_bvals)
+
+    xb = -np.log(np.arange(1, len(unique_bvals)))
+
+    # Reshape xb for linear regression
+    X = xb[:len(S_log)].reshape(-1, 1)
+    y = S_log
+
+    # Fit linear model
+    model = LinearRegression()
+    model.fit(X, y)
+    P = model.coef_[0]
+    V = model.intercept_
+
+    return P, V
+
+
+def compute_P_map(data, bvals, wm_threshold=0.5, b0_thresh=10):
```

Comment:
same comments as above concerning ` b0_thresh` and keyword only arguments.

Also, p should be lowercase in the function name